core/server/worker/src/test/java/alluxio/worker/block/BlockRWLockPoolTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.block;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.BrokenBarrierException;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.TimeUnit;
+
+public class BlockRWLockPoolTest {
+  /**
+   * To test the number of remaining resources in the pool. We report this value as a metric.
+   * pool size of 10, 10 threads acquire locks and release early
+   * while another 10 threads acquire locks but release late.
+   * in the first checkpoint, the remaining resource should be 0,
+   * in the second checkpoint,
+   * the remaining resource should be the size of the pool as we returned all locks.
+   */
+  @Test
+  public void testMetricReporting() throws BrokenBarrierException, InterruptedException {
+    final int numIterations = 10;
+    final int numLocksEarlyRelease = 10;
+    final int numLocksLateRelease = 10;
+    final int poolCapacity = 10;
+    Random r = new Random();
+    for (int i = 0; i < numIterations; ++i) {
+      BlockRWLockPool pool = new BlockRWLockPool(poolCapacity);
+      List<Thread> threads = new ArrayList<>();
+      final CyclicBarrier barrier1 =
+          new CyclicBarrier(numLocksEarlyRelease + numLocksLateRelease + 1);
+      final CyclicBarrier barrier2 = new CyclicBarrier(numLocksLateRelease + 1);
+      for (int j = 0; j < numLocksEarlyRelease; ++j) {
+        threads.add(new Thread(
```

Comment:
Would a thread pool do a better job than manually creating and destroying threads?